name: E2E Python Tests

on:
  pull_request:
    types: [opened, synchronize, reopened, ready_for_review]

permissions:
  contents: read

defaults:
  run:
    working-directory: ./python

jobs:
  e2e:
    runs-on: ubuntu-latest

    strategy:
      matrix:
        python: ["3.10", "3.11", "3.12"]

    steps:
      - name: Checkout
        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2

      - name: Set up Python ${{ matrix.python }}
        uses: actions/setup-python@8d9ed9ac5c53483de85588cdf95a591a75ab9f55 # v5.5.0
        with:
          python-version: ${{ matrix.python }}

      - name: Install uv
        run: pip install uv

      - name: Setup environment and install dependencies
        run: |
          make setup
          make verify-env
          make install

      - name: Run E2E tests
        env:
          CDP_API_KEY_ID: ${{ secrets.CDP_API_KEY_ID_NEW }}
          CDP_API_KEY_SECRET: ${{ secrets.CDP_API_KEY_SECRET_NEW }}
          CDP_WALLET_SECRET: ${{ secrets.CDP_WALLET_SECRET_NEW }}
          DISABLE_CDP_ERROR_REPORTING: true
        run: uv run pytest cdp/test/test_e2e.py -v
